### PR TITLE
feat(project): introduce cache manager for project (#16740)

### LINKED ISSUE #16740: Cache layer for project.
Implement cache layer for project.

### PR DESCRIPTION
**Pull request by chlins**
Implement cached manager for resource project and migrate project.Mgr
to pkg.ProjectMgr.

Signed-off-by: chlins <chenyuzh@vmware.com>

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #16740 

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/pkg/factory.go [hunk 1]
```diff
@@ -19,7 +19,19 @@ import (
...
+	cachedArtifact "github.com/goharbor/harbor/src/pkg/cached/artifact/redis"
+	cachedProject "github.com/goharbor/harbor/src/pkg/cached/project/redis"
+	"github.com/goharbor/harbor/src/pkg/project"
+)
+
+type initManagerFunc func(cacheEnable bool)
+
+var (
+	// initManagerFuncs is the registry for init manager func.
+	initManagerFuncs = []initManagerFunc{
```
Comment (stonezdj) [inline src/pkg/factory.go:31]: Not clear about the intention of initManagerFuncs.  A slice will make us consider it is configurable,  for example if I want to disable cache in ArtifactManager, just removing the initArtifactManager from slice, but the current code doesn't work in this way.
Suggest to remove this initManagerFuncs and call it in the init function of current package
```
func init(){
   initArtifactManager(true)
   initProjectManager(true)
}
```

File: src/pkg/cached/project/redis/manager.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	return nil
+}
+
+func (m *manager) Get(ctx context.Context, idOrName interface{}) (*models.Project, error) {
+	var (
+		key string
+		err error
+	)
+
+	id, ok := idOrName.(int64)
```
Comment (wy65701436) [inline src/pkg/cached/project/redis/manager.go:101]: update & leverage the common/util.ParseProjectIDOrName()?

**Review by wy65701436 — APPROVED** [review]
lgtm

File: src/pkg/factory.go (hunk 1 above)
Comment (chlins) [reply src/pkg/factory.go]: updated

File: src/pkg/cached/project/redis/manager.go (hunk 2 above)
Comment (chlins) [reply src/pkg/cached/project/redis/manager.go]: updated

**Review by stonezdj — APPROVED** [review]

**Pull request merged**